### ISSUE TITLE
*log.safeError: conn_executor.go:668: panic while executing 1 statements: SELECT _, _ FROM _ WHERE _ = $1: caused...

### ISSUE THREAD

**Issue by jordanlewis** [cockroachdb/cockroach#30535]
https://sentry.io/cockroach-labs/cockroachdb/issues/698501126/

```
github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go in assignPlaceholders at line 11718
github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go in assignPlaceholders at line 11443
github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go in assignPlaceholders at line 11416
github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go in assignPlaceholders at line 11179
github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.og.go in assignPlaceholders at line 11183
github.com/cockroachdb/cockroach/pkg/sql/opt/norm/factory.go in AssignPlaceholders at line 167
github.com/cockroachdb/cockroach/pkg/sql/plan.go in makeOptimizerPlan at line 465
github.com/cockroachdb/cockroach/pkg/sql/planner.go in optionallyUseOptimizer at line 534
github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go in dispatchToExecutionEngine at line 753
github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go in execStmtInOpenState at line 403
github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go in execStmt at line 95
github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go in run at line 1136
github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go in ServeConn at line 416
github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go in func3 at line 267
```

**Comment by jordanlewis** [comment]
#30535

**Comment by jordanlewis** [comment]
Rather, #30538.

**Comment by andy-kimball** [comment]
Fixed by pull request #30860.
